deeptime/src/include/deeptime/markov/msm/tram/connected_set.h

```diff
@@ -196,13 +196,27 @@ TransitionVector findStateTransitions(const std::optional<DTrajs> &ttrajs,
     std::vector<IndexList> thermStateIndicesFromPerThread(nThreads);
     std::vector<IndexList> thermStateIndicesToPerThread(nThreads);
 
+    auto nTrajs = dtrajs.size();
+
     // threads don't like references
-    auto dtrajsPtr = &dtrajs;
-    auto ttrajsPtr = &ttrajs;
-    auto biasMatricesPtr = &biasMatrices;
-    auto stateCountsBuf = stateCounts.template unchecked<2>();
+    std::vector<ArrayBuffer<DTraj, 1>> dtrajBuffers (dtrajs.begin(), dtrajs.end());
+    auto dtrajsPtr = dtrajBuffers.data();
+
+    ArrayBuffer<DTraj, 1> *ttrajsPtr = nullptr;
+    std::vector<ArrayBuffer<DTraj, 1>> ttrajBuffers;
+
+    if (ttrajs) {
+        ttrajBuffers = std::vector<ArrayBuffer<DTraj, 1>> ((*ttrajs).begin(), (*ttrajs).end());
+        ttrajsPtr = ttrajBuffers.data();
```

Comment:
you can always set `ttrajsPtr = ttrajBuffers.data()` by just moving the initialization of the ptr to after the if statement and using `const auto* ttrajsPtr = ttrajBuffers.data()`.